### PR TITLE
OKTA-369793 Add ThreatInsight API to OpenAPI spec

### PR DESCRIPTION
**Pull request by andriizhegurov-okta**
This endpoint is now GA and needs to be added to the spec: https://developer.okta.com/docs/reference/api/threat-insight/

### CODE REVIEW

**Review by bretterer — CHANGES_REQUESTED** [review]
Because this is a new `Tag`, we are going to need to include x-okta-crud methods to generate this as a Resource object. See items like `Application` or `User` for what i'm talking about here. We will also need `x-okta-operations`

**Comment by andriizhegurov-okta** [comment]
@bretterer I added `x-okta-crud` tag, `x-okta-operations` doesn't seem to apply for this case. Did you mean we need to add an empty  `x-okta-operations` tag?

File: dist/spec.yaml [hunk 1]
```diff
@@ -4896,6 +4896,46 @@ paths:
...
+            $ref: '#/definitions/ThreatInsightConfiguration'
+      security:
+        - api_token: []
+      tags:
+        - ThreatInsight
+    post:
+      consumes:
+        - application/json
+      description: Updates ThreatInsight configuration
+      operationId: updateThreatInsightConfiguration
```
Comment (laura-rodriguez) [inline dist/spec.yaml:4921]: If the client is already named `ThreatInsight` then we might want to rename this method to `updateConfiguration`, this way we'll have this: `ThreatInsightClient.UpdateConfiguration()`.

File: dist/spec.yaml [hunk 2]
```diff
@@ -11824,6 +11864,38 @@ definitions:
...
+      excludeZones:
+        items:
+          type: string
+        type: array
+      lastUpdated:
+        format: date-time
+        readOnly: true
+        type: string
+    x-okta-crud:
+      - alias: read
```
Comment (laura-rodriguez) [inline dist/spec.yaml:11889]: We don't use `x-okta-crud` in .NET, but I'm not sure we use `read`/GET in other entities. Can you check with someone that utilizes the crud section? Maybe @bretterer ?

File: dist/spec.yaml [hunk 3]
```diff
@@ -4896,6 +4896,46 @@ paths:
       summary: Update SMS Template
       tags:
         - Template
+  /api/v1/threats/configuration:
+    get:
+      consumes:
+        - application/json
+      description: Gets current ThreatInsight configuration
+      operationId: getThreatInsightConfiguration
```
Comment (laura-rodriguez) [inline dist/spec.yaml:4904]: I'd rename this to `getCurrentConfiguration`

**Review by laura-rodriguez — COMMENTED** [review]
I have a few suggestions, please review my comments.

File: dist/spec.yaml (hunk 2 above)
Comment (andriizhegurov-okta) [reply dist/spec.yaml:11889]: This is waiting for @bretterer to review

**Comment by andriizhegurov-okta** [comment]
@laura-rodriguez I renamed the IDs you've suggested

File: dist/spec.yaml (hunk 2 above)
Comment (arvindkrishnakumar-okta) [reply dist/spec.yaml:11889]: `x-okta-crud` section looks good to me. I had verified Java SDK code generation with it. cc: @sergiishamrai-okta

**Review by arvindkrishnakumar-okta — APPROVED** [review]
🚢

**Review by bretterer — APPROVED** [review]

**Pull request merged**